README.md
**Dont forget to add ![/image/star.png](https://github.com/mamin27/ecomet_i2c_raspberry_tools/blob/master/python_test_scripts/display/images/star.png) if you were satisfy with the software!**

# H616-mangopi (version 1.2)
DTS file for setting MQ-Quad / MCore-H616 board

[MQ-Quad H616](https://mangopi.org/mqquad)
# Compile DTS fit to DTB binary and reboot

```ps
sudo apt-get install sunxi-tools device-tree-compiler rpi.gpio-common

cd /boot/dtb/allwinner/
dtc -I dts -O dtb -f sun50i-h616-biqu.dts -o sun50i-h616-biqu.dtb
```
# OPi.GPIO package for H616 board (my clone of origin library)
* current version 0.5.4 (added some not applied pull-request from origin https://github.com/rm-hull/OPi.GPIO project & my changes)
* this version is not distributed by pypi.org, I respect owner of the project.
* Added mangopi library
* How to compile lib:
```ps
cd OPi.GPIO
sudo bin/make_release.sh
cd dist
pip3 install OPi.GPIO-0.5.4-py2.py3-none-any.whl
```
* test of flashing LED
```ps
./led_mq_quad.py
```

### features:
* I2C (/dev/i2c-1)
* SPI
* I2S0 TBD (Look at issue #1 and help with investigatio)
* activate sensor-led TBD


```ps
comet@mangopi:/boot/dtb/allwinner$ i2cdetect -y 1
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:                         -- -- -- -- -- -- -- --
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
20: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
40: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
50: 50 -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
60: -- -- 62 -- -- -- -- -- -- -- -- -- -- -- -- --
70: 70 -- -- -- -- -- -- --
```


OPi.GPIO/CHANGES.rst
ChangeLog
---------

+------------+---------------------------------------------------------------------+------------+
| Version    | Description                                                         | Date       |
+============+=====================================================================+============+
| **0.5.4**  | * Added MangoPi MG-QUAD                                             | 2023/10/02 |
|            | * Added OrangePi Zero3, Zero 2w pin mappings                        |            |
|            | * Onboard LED support                                               |            |
|------------+---------------------------------------------------------------------+------------|
| **0.5.3**  | * Add pin mappings for Radxa Zero                                   | 2022/06/20 |
|            | * Added OrangePi Zero2 pin mappings                                 |            |
+------------+---------------------------------------------------------------------+------------+
| **0.5.2**  | * Add full 40-pin header for Orange Pi 4(B)                         | 2021/10/22 |
+------------+---------------------------------------------------------------------+------------+
| **0.5.1**  | * Updated orange pi 4B mappings                                     | 2021/08/19 |
+------------+---------------------------------------------------------------------+------------+
| **0.5.0**  | * Added Hardware PWM Control                                        | 2021/04/25 |
|            | * Add support for the NanoPi M4 family                              |            |
|            | * Add pin mapping for orange pi 4 & 4B models                       |            |
|            | * Add rockpi/rockpi s packages added                                |            |
|            | * Fix OrangePi PC BCM Pin map                                       |            |
+------------+---------------------------------------------------------------------+------------+
| **0.4.0**  | * Added various new pin mappings for Orange Pi variants             | 2019/10/21 |
+------------+---------------------------------------------------------------------+------------+
| **0.3.6**  | * Expose pullup resistor constants to API users                     | 2019/01/14 |
|            | * Add pin mappings for OrangePi Prime                               |            |
+------------+---------------------------------------------------------------------+------------+
| **0.3.5**  | * Added OrangePi Lite & One pin mappings                            | 2018/11/10 |
+------------+---------------------------------------------------------------------+------------+
| **0.3.4**  | * Added more OrangePi and NanoPi pin mappings                       | 2018/09/16 |
+------------+---------------------------------------------------------------------+------------+
| **0.3.3**  | * Added waits for UDEV rules                                        | 2018/09/15 |
+------------+---------------------------------------------------------------------+------------+
| **0.3.2**  | * Added OrangePi PC pin mappings                                    | 2018/03/04 |
+------------+---------------------------------------------------------------------+------------+
| **0.3.1**  | * Added NanoPi DUO pin mappings                                     | 2018/01/01 |
+------------+---------------------------------------------------------------------+------------+
| **0.3.0**  | * Added alternate pin mappings                                      | 2017/12/31 |
+------------+---------------------------------------------------------------------+------------+
| **0.2.5**  | * sysfs: set output() value to 0 or 1                               | 2017/07/15 |
+------------+---------------------------------------------------------------------+------------+
| **0.2.4**  | * Add compatibility for pull up/down and bouncetime params          | 2017/05/27 |
+------------+---------------------------------------------------------------------+------------+
| **0.2.3**  | * Make worker threads daemonic (can't exit otherwise)               | 2017/05/26 |
+------------+---------------------------------------------------------------------+------------+
| **0.2.2**  | * ``GPIO.setup()`` catches IOError and re-exports                   | 2017/03/28 |
+------------+---------------------------------------------------------------------+------------+
| **0.2.1**  | * Minor bug fixes                                                   | 2017/03/14 |
|            | * Additional tests                                                  |            |
+------------+---------------------------------------------------------------------+------------+
| **0.2.0**  | * Added edge detection and eventing                                 | 2017/03/14 |
+------------+---------------------------------------------------------------------+------------+
| **0.1.0**  | * Initial version                                                   | 2017/03/11 |
+------------+---------------------------------------------------------------------+------------+


OPi.GPIO/orangepi/zero2w.py
# -*- coding: utf-8 -*-
# Copyright (c) 2022 sillo01
# See LICENSE.md for details.

"""
Alternative pin mappings for Orange PI Zero2
(see https://linux-sunxi.org/File:Orange_Pi_Zero2_H616_Schematic_v1.3.pdf)
Usage:
.. code:: python
   import orangepi.zero2w
   from OPi import GPIO
   GPIO.setmode(orangepi.zero2w.BOARD)
 +------+-----+----------+--------+---+  ZERO2W  +---+--------+----------+-----+------+
 | GPIO | wPi |   Name   |  Mode  | V | Physical | V |  Mode  | Name     | wPi | GPIO |
 +------+-----+----------+--------+---+----++----+---+--------+----------+-----+------+
 |      |     |     3.3V |        |   |  1 || 2  |   |        | 5V       |     |      |
 |  264 |   0 |    SDA.1 |    OFF | 0 |  3 || 4  |   |        | 5V       |     |      |
 |  263 |   1 |    SCL.1 |    OFF | 0 |  5 || 6  |   |        | GND      |     |      |
 |  269 |   2 |     PWM3 |    OFF | 0 |  7 || 8  | 0 | ALT2   | TXD.0    | 3   | 224  |
 |      |     |      GND |        |   |  9 || 10 | 0 | ALT2   | RXD.0    | 4   | 225  |
 |  226 |   5 |    TXD.5 |    OFF | 0 | 11 || 12 | 0 | OFF    | PI01     | 6   | 257  |
 |  227 |   7 |    RXD.5 |    OFF | 0 | 13 || 14 |   |        | GND      |     |      |
 |  261 |   8 |    TXD.2 |    OFF | 0 | 15 || 16 | 0 | OFF    | PWM4     | 9   | 270  |
 |      |     |     3.3V |        |   | 17 || 18 | 0 | OFF    | PH04     | 10  | 228  |
 |  231 |  11 |   MOSI.1 |    OFF | 0 | 19 || 20 |   |        | GND      |     |      |
 |  232 |  12 |   MISO.1 |    OFF | 0 | 21 || 22 | 0 | OFF    | RXD.2    | 13  | 262  |
 |  230 |  14 |   SCLK.1 |    OFF | 0 | 23 || 24 | 0 | OFF    | CE.0     | 15  | 229  |
 |      |     |      GND |        |   | 25 || 26 | 0 | ALT3   | CE.1     | 16  | 233  |
 |  266 |  17 |    SDA.2 |    OFF | 0 | 27 || 28 | 0 | OFF    | SCL.2    | 18  | 265  |
 |  256 |  19 |     PI00 |    OFF | 0 | 29 || 30 |   |        | GND      |     |      |
 |  271 |  20 |     PI15 |    OFF | 0 | 31 || 32 | 0 | OFF    | PWM1     | 21  | 267  |
 |  268 |  22 |     PI12 |    OFF | 0 | 33 || 34 |   |        | GND      |     |      |
 |  258 |  23 |     PI02 |    OFF | 0 | 35 || 36 | 1 | OUT    | PC12     | 24  | 76   |
 |  272 |  25 |     PI16 |    OFF | 0 | 37 || 38 | 0 | OFF    | PI04     | 26  | 260  |
 |      |     |      GND |        |   | 39 || 40 | 0 | OFF    | PI03     | 27  | 259  |
 +------+-----+----------+--------+---+----++----+---+--------+----------+-----+------+
 | GPIO | wPi |   Name   |  Mode  | V | Physical | V |  Mode  | Name     | wPi | GPIO |
 +------+-----+----------+--------+---+  ZERO2W  +---+--------+----------+-----+------+
"""

# Orange Pi Zero 2w physical board pin to GPIO pin
# PH1(10) & PH2(11) for debug console default
BOARD = {
    3: 264,  # PI8 / TWI1_SDA
    5: 263,  # PI7 / TWI1_SCL
    7: 269,  # PI13 / PWM3 / UART4_TX
    8: 224,  # PH0 / UART0_TX
    10: 225,  # PH1 / UART0_RX
    11: 226,  # PH2 / UART5_TX
    12: 257,  # PI1
    13: 227,  # PH3 / UART5_RX
    15: 261,  # PI5 / TWI0_SCL / UART2_TX
    16: 270,  # PI14 / PWM4 / UART4_RX
    18: 228,  # PH4
    19: 231,  # PH7 / SPI1_MOSI
    21: 232,  # PH8 / SPI1_MISO
    22: 262,  # PI6 / TWI0_SDA / UART2_RX
    23: 230,  # PH6 / SPI1_CLK
    24: 229,  # PH5 / SPI1_CS0
    26: 233,  # PH9 / SPI1_CS1
    27: 266,  # PI10 / TWI2_SDA / UART3_RX
    28: 265,  # PI9 / TWI2_SCL / UART3_TX
    29: 256,  # PI0
    31: 271,  # PI15
    32: 265,  # PI11 / PWM1
    33: 268,  # PI12 / PWM2
    35: 258,  # PI2
    36: 76,  # PC12
    37: 272,  # PI16
    38: 260,  # PI4
    40: 259,  # PI3
}

# BCM mapping not identified yet, keeping it for compatibility
BCM = BOARD